modin/core/dataframe/pandas/metadata/dtypes.py

```diff
@@ -294,6 +294,11 @@ def set_index(
         Calling this method on a descriptor that returns ``None`` for ``.columns_order``
         will result into information lose.
         """
+        if len(new_index) != len(set(new_index)):
```

Comment:
Would it be better to use method `nunique`?

Reply:
[`np.unique` struggles to handle MultiIndex objects](https://github.com/modin-project/modin/pull/6663#discussion_r1396302560)

Reply:
`new_index.unique()` might also not always work as sometimes it can be a python list here

Reply:
> `new_index.unique()` might also not always work as sometimes it can be a python list here

Should we update the docstring for that?

Reply:
yep, docstring seems to be incorrect, will update in some future PR